### PR TITLE
Add a few testing helper utilities to `pueblo.testing`

### PR DESCRIPTION
**Pull request by amotl**
## About

Classical yak shaving.

## Details

A few generalized testing helpers to avoid code duplication within the educational repository [cratedb-examples](https://github.com/crate/cratedb-examples), originally conceived at [^1]. We need those utilities at many more spots, so we needed to find a good place to store them.

[^1]: https://github.com/crate/cratedb-examples/blob/main/topic/machine-learning/llm-langchain/test.py

/cc @andnig

### CODE REVIEW

File: pueblo/testing/folder.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+from pathlib import Path
+
+
+def list_files(path: Path, pattern: str):
```
Comment (matriv) [inline pueblo/testing/folder.py:4]: minor: This method seems to not be tested separately, (of course it's used by the other 2 which are tested), maybe add a test for this one as well?

Comment (amotl) [reply pueblo/testing/folder.py:4]: For now, I think will be good enough. I don't want to over-engineer the code here. There are lots of opportunities for making the tests even stronger, currently there are gaps, specifically on evaluating exception handling. Maybe I will spend some extra minutes on it, on behalf of a different iteration. Right now, I would like to move on.

**Review by matriv — APPROVED** [review]

File: pueblo/testing/folder.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+
+def list_python_files(path: Path, pattern: str = "*.py"):
+    """
+    Enumerate all regular Python files found in given directory.
+    """
+    pyfiles = []
+    for item in list_files(path, pattern):
+        if item.name in ["conftest.py"] or item.name.startswith("test"):
+            continue
```
Comment (amotl) [inline pueblo/testing/folder.py:26]: Currently untested, so this is a candidate for improving the test cases.

File: pueblo/testing/snippet.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    test cases, and run them.
+
+    Not using `NBRegressionFixture`, because it would manually need to be configured.
+    """
+    from _pytest._py.path import LocalPath
+    from pytest_notebook.plugin import pytest_collect_file
+
+    tests = pytest_collect_file(LocalPath(filepath), request.node)
+    if not tests:
+        raise ValueError(f"No tests collected from notebook: {filepath}")
```
Comment (amotl) [inline pueblo/testing/snippet.py:43]: Currently untested, so this is a candidate for improving the test cases.

File: pueblo/testing/folder.py (hunk 1 above)
Comment (matriv) [reply pueblo/testing/folder.py:4]: thank you!

**Pull request merged**